tests/gcloud-testing-core/src/main/java/com/google/cloud/runtime/jetty/testing/HttpUrlUtil.java

```diff
@@ -0,0 +1,105 @@
+package com.google.cloud.runtime.jetty.testing;
+
+import org.eclipse.jetty.toolchain.test.IO;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class HttpUrlUtil {
+  private static final Logger log = Logger.getLogger(HttpUrlUtil.class.getName());
+
+  /**
+   * Wait for a server to be "up" by requesting a specific GET resource
+   * that should be returned in status code 200.
+   * <p>
+   * This will attempt a check for server up.
+   * If any result other then response code 200 occurs, then
+   * a 2s delay is performed until the next test.
+   * Up to the duration/timeunit specified.
+   * </p>
+   *
+   * @param uri      the URI to request
+   * @param duration the time duration to wait for server up
+   * @param unit     the time unit to wait for server up
+   */
+  public static void waitForServerUp(URI uri, int duration, TimeUnit unit) {
+    System.err.println("Waiting for server up: " + uri);
+    boolean waiting = true;
+    long expiration = System.currentTimeMillis() + unit.toMillis(duration);
+    while (waiting && System.currentTimeMillis() < expiration) {
+      try {
+        System.out.print(".");
+        HttpURLConnection http = (HttpURLConnection) uri.toURL().openConnection();
+        http.setRequestProperty("User-Agent", "jetty-runtime/gcloud-testing-core(server-up)");
+        int statusCode = http.getResponseCode();
+        if (statusCode != HttpURLConnection.HTTP_OK) {
+          log.log(Level.FINER, "Waiting 2s for next attempt");
+          TimeUnit.SECONDS.sleep(2);
+        } else {
+          waiting = false;
+        }
+      } catch (MalformedURLException e) {
+        Assert.fail("Invalid URI: " + uri.toString());
```

Comment:
Why not just remove this catch block or wrap it into an IllegalArgumentException? This will allow us to get rid of the junit-compile dependency and give more flexibility to the caller.